### PR TITLE
Account registration, login & navigation menu

### LINKED ISSUE #43: Account Registration
As a user, I need to be able to register myself as a driver in the app with my individual information & emergency contacts.

**Acceptance Criteria**
- [x] Register option is visible to a new user in navigation side bar
- [x] When "Register" is clicked, it redirects to a new page 
- [x] Entry fields for Name, Age, Email Address, Password, and Emergency Contacts
- [x] Password field hides entered information
- [x] Prominent Register button conforming to app color scheme
- [x] Connected to Firebase authentication service

### LINKED ISSUE #45: Navigation Menu
As a user I want to have a clutter free experience while using the app to control the car.

**Acceptance criteria:**
- [x] A toggle for the navigation menu is visible at the top left of the application
- [x] Navigation menu has a header with our logo
- [x] Core items are visible below the header whilst navigation menu is toggled on
- [x] Menu items are shown or hidden depending on the login status of the user
- [x] An icon exists for each navigation menu item facilitating instant recognition and subsequently aiding with navigation and tasks

### PR DESCRIPTION
**Pull request by sepehrmoradian**
Closes #43 & Closes #45 

A navigation menu has been implemented to clear up several items from the main driving screen. Firebase authentication integrated to facilitate account registration and login.

- Integrated navigation menu to create a clutter free driving experience
- Added the ability to create a user account and login, which leads to more customisation further down the line
- Sanity checks for forms
- Confirmation / Error Toasts
- A straightforward profile page showing relevant information
- Reworked adding and viewing emergency contacts into the new menu
- Based on login status certain menu items are either visible or hidden
- App icon to make it look swanky
- Safety features toggle button sizing to unify overall look and feel of the app
- Integrated previously implemented action bar buttons pertaining to calling and messaging emergency contacts in tandem with navigation menu

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MailBot/MailSender.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package safetyfirst.androidapp.safetyfirstcontroller.MailBot;
 
+//import java.net.PasswordAuthentication;
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MailBot/MailSender.java]: Why is this commented out?

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java [hunk 2]
```diff
@@ -145,6 +178,85 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean b) {//Publis
         });
     }
 
+    @SuppressLint("NonConstantResourceId")
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]:
```suggestion
    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
```
maybe a more explanatory variable name here

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java [hunk 3]
```diff
@@ -145,6 +178,85 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean b) {//Publis
...
+                        new HomeFragment()).commit();
+                Toast.makeText(MainActivity.this, "Logged Out", Toast.LENGTH_LONG).show();
+        }
+        drawerLayout.closeDrawer(GravityCompat.START);
+        return true;
+    }
+
+    public void hideItemLogged() {
+        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+        Menu nav_Menu = navigationView.getMenu();
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Naming conventions

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java [hunk 4]
```diff
@@ -145,6 +178,85 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean b) {//Publis
...
+        nav_Menu.findItem(R.id.profile).setVisible(true);
+        nav_Menu.findItem(R.id.login).setVisible(false);
+        nav_Menu.findItem(R.id.registerUser).setVisible(false);
+        nav_Menu.findItem(R.id.logOut).setVisible(true);
+
+    }
+
+    public void hideItemDefault() {
+        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+        Menu nav_Menu = navigationView.getMenu();
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Naming conventions

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                }
+                firebaseAuth.signInWithEmailAndPassword(email, password)
+                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                            @Override
+                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                if (task.isSuccessful()) {
+                                    hideKeyboardFrom(requireContext(), view);
+
+                                    NavigationView navigationView = (NavigationView) requireActivity().findViewById(R.id.nav_view);
+                                    Menu nav_Menu = navigationView.getMenu();
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java]: naming conventions

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/ViewContactFragment.java [hunk 6]
```diff
@@ -1,43 +1,42 @@
...
+import safetyfirst.androidapp.safetyfirstcontroller.R;
 
-public class ContactList extends AppCompatActivity {
-
-
+public class ViewContactFragment extends Fragment {
     //Database
     ArrayAdapter customerArrayAdapter;
     DataBaseHelper dataBaseHelper;
+    //ListView lv_contactList;
```
Comment (Valkama1) [inline android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/ViewContactFragment.java]: Why is this commented out?

File: android/app/src/main/res/layout/nav_header.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    android:padding="16dp"
+    android:theme="@style/ThemeOverlay.AppCompat.Light">
+
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:src="@mipmap/car_safe_round" />
+
+    <TextView
+        android:id="@+id/text1"
```
Comment (Valkama1) [inline android/app/src/main/res/layout/nav_header.xml]: text1? Please change to a more descriptive name

File: android/app/src/main/res/menu/first_menu.xml [hunk 8]
```diff
@@ -2,22 +2,6 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (Valkama1) [inline android/app/src/main/res/menu/first_menu.xml:2]: Is this file needed?

**Review by Valkama1 — CHANGES_REQUESTED** [review]
Looks pretty good for such a huge PR, check the comments I've made and try to split PRs like this one into multiple in the future. Also, it would be good to see more comments as most of the code has none.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MailBot/MailSender.java (hunk 1 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MailBot/MailSender.java]: Removed now, I used a different import with the same functionality in it's place as there were some issues relating to the commented out import.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 2 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Suggested change implemented.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 3 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Here it's referencing an "id" in a XML file, which does not have the same naming conventions as java.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 4 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Refer to previously resolved conversation.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java (hunk 5 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java]: Refer to previously resolved conversation.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/ViewContactFragment.java (hunk 6 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/ViewContactFragment.java]: Has been removed, the variable in use is written below this.

File: android/app/src/main/res/menu/first_menu.xml (hunk 8 above)
Comment (sepehrmoradian) [reply android/app/src/main/res/menu/first_menu.xml:2]: Yes, it's the icons that appear in the action bar.

**Review by Valkama1 — DISMISSED** [review]

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 3 above)
Comment (Valkama1) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: nav_Menu in this line is a variable name in Java.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 4 above)
Comment (Valkama1) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Same as previous convo

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java (hunk 5 above)
Comment (Valkama1) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java]: Same again

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 3 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Ah yes, my bad can't believe I missed that. What a world we live in.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java (hunk 4 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/MainActivity.java]: Fixed.

File: android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java (hunk 5 above)
Comment (sepehrmoradian) [reply android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java]: Fixed. Thanks.

**Review by Valkama1 — APPROVED** [review]
Looks good now, great work boiiii
